infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -394,20 +371,21 @@ public void clearLocalStreams() {
 
         log.debug("Total of {} streams were replicated from active out of {}, sequencer query for {}, streamsToQuery={}",
                 replicatedStreamIds.size(), streamViewMap.size(), streamsToQuery.size(), streamsToQuery);
-        Set<UUID> streamsToClear = new HashSet<>();
         TokenResponse tokenResponse = rt.getSequencerView().query(streamsToQuery.toArray(new UUID[0]));
+        Set<UUID> locallyUpdated = new HashSet<>();
         streamsToQuery.forEach(streamId -> {
             if (tokenResponse.getStreamTail(streamId) != Address.NON_EXIST) {
-                streamsToClear.add(streamId);
+                locallyUpdated.add(streamId);
             }
         });
 
-        if (!streamsToClear.isEmpty()) {
-            log.debug("Clear streams with local writes, total={}, streams={}", streamsToClear.size(), streamsToClear);
-            clearStreams(streamsToClear);
+        if (!locallyUpdated.isEmpty()) {
+            log.debug("Clear streams with local writes, total={}, streams={}", locallyUpdated.size(), locallyUpdated);
+            streamsToClear.addAll(locallyUpdated);
         } else {
             log.debug("No local written streams were found, nothing to clear.");
         }
+        clearStreams(streamsToClear);
```

Comment:
Just realized, now that we are iterating over the whole streams to replicate,I thought the pre-adding `streamsToClear` is unnecessary? (the replicated streams are guaranteed to be included in the streams to replicate set). So maybe just check all the streams with local writes and throw them into `clearStreams()`?

Also, I thought there is already a check for merge-only streams in `clearStreams()` method

Reply:
yeah, I kept the locallyUpdated stream mainly for logging purpose..thinking that knowing which stream had data coming from active and which had only local updates could be helpful while tracing the streams.

Can remove that if everyone thinks its not necessary and keep directly throwing to clearStreams()